newsrevenuehub/rev-engine
# [WIP] NRH 242:  facebook pixel integration

#### What's this PR do?

- Hub admins can add a facebook pixel id to a revenue program
- Donation pages for that revenue program will send a pageview event to facebook pixel
- Thank you pages for that revenue program will send a pageview event, as well as tracking "Donate" and "Purchase" events

#### How should this be manually tested?

- Add valid facebook pixel id to a revenue program
- Open a browser tab, and open developer tools
- In same browser tab, go to a live donation page for the rev program
- Confirm a pageview event fires to facebook in network tab of dev tools. The url will be like https://connect.facebook.net/signals/config/<ORG_FACEBOOK_PIXEL_ID>?v=2.9.44&r=stable (substituting a real value for ORG_FACEBOOK_PIXEL_ID).
- Successfully complete a donation and get sent to the thank you page
- Confirm a pageview event fires to facebook in network tab of dev tools. The url will be like https://connect.facebook.net/signals
- [details on confirming event trackers for conversion]

#### Do any changes need to be made before deployment to staging? production? (adding environment variables, for example)?

No
